src/wallet/rpcwallet.cpp

```diff
@@ -1162,6 +1159,10 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
             if (!ExtractDestination(txout.scriptPubKey, address))
                 continue;
 
+            if (fFilterAddress && !(filterAddress.Get() == address)) {
```

Comment:
Maybe we should be storing `filterAddress.Get()` above rather than a `CBitcoinAddress`?